client/image_pull.go

```diff
@@ -2,19 +2,46 @@ package client
 
 import (
 	"context"
+	"encoding/json"
+	"errors"
 	"io"
+	"iter"
 	"net/url"
 	"strings"
 
 	cerrdefs "github.com/containerd/errdefs"
 	"github.com/distribution/reference"
+	"github.com/moby/moby/client/pkg/jsonmessage"
 )
 
+type ImagePullResponse struct {
+	io.ReadCloser
```

Comment:
```suggestion
	rd io.ReadCloser
```

We should probably make it a concrete field and then implementing the `io.ReadCloser` for the `ImagePullResponse`.

This will also allow us to wrap `Close` method so we don't close the underlying reader twice in cases like:

```go
out, err := ImagePull(...)
...

defer out.Close()

for i in out.JSONMessages() {
...
}
...
```